### PR TITLE
Adding WCs per process

### PR DESCRIPTION
**Pull request by sscruz**
Adding the possibility to use a different list of samples per process. The list of selected WCs is inputed through a json file stored now in `Fitter/hist_files/selectedWCs.txt`
This json's been constructed using https://github.com/TopEFT/topcoffea/pull/125

### CODE REVIEW

File: Fitter/hist_files/selectedWCs.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+{"tHq": ["ctp", "cptb", "cQq13", "cbW", "cpQ3", "ctW", "cQq83", "ctG"], "tllq": ["cpt", "cptb", "cQlMi", "cQl3i", "ctlTi", "ctli", "cQq13", "cbW", "cpQM", "cpQ3", "ctei", "cQei", "ctW", "ctlSi", "cQq83", "ctZ", "ctG"], "ttH": ["cpt", "ctp", "cptb", "cQq81", "cQq11", "ctq8", "ctq1", "cQq13", "cbW", "cpQM", "cpQ3", "ctW", "cQq83", "ctZ", "ctG"], "ttll": ["cpt", "cptb", "cQlMi", "cQq81", "cQq11", "cQl3i", "ctq8", "ctlTi", "ctq1", "ctli", "cQq13", "cbW", "cpQM", "cpQ3", "ctei", "cQei", "ctW", "ctlSi", "cQq83", "ctZ", "ctG"], "ttlnu": ["cpt", "ctp", "cQlMi", "cQq81", "cQq11", "cQl3i", "ctq8", "ctlTi", "ctq1", "ctli", "cQq13", "cpQM", "cpQ3", "ctW", "ctlSi", "cQq83", "ctZ", "ctG"], "tttt": ["cpt", "ctp", "cptb", "cQq81", "cQq11", "ctq8", "ctq1", "cQq13", "cbW", "cpQM", "cpQ3", "ctW", "cQq83", "ctZ", "ctG", "ctt1", "cQt1", "cQt8", "cQQ1"]}
```
Comment (bryates) [inline Fitter/hist_files/selectedWCs.txt:1]: Is this something we'll have to generate each time? I suppose you picked out the WCs that each process is most sensitive to.
Also, should we rename it to `selectedWCs.json`?

File: Fitter/python/AnomalousCouplingEFTNegative.py [hunk 2]
```diff
@@ -289,24 +178,27 @@ def doParametersOfInterest(self):
...
         if any( process.startswith(x) for x in self.sgnl_known):
-            if self.sm_re.search(process): 
-                return "func_sm"
+            if self.sm_re.search(process):
+                match=self.sm_re.search(process)
+                return "func_%s_sm"%(match.group('proc'))
             elif self.lin_re.search(process): 
                 match=self.lin_re.search(process)
-                return "func_sm_linear_quadratic_"+ match.group('c1')
+                return "func_%s_sm_linear_quadratic_"%(match.group('proc'))+ match.group('c1')
```
Comment (bryates) [inline Fitter/python/AnomalousCouplingEFTNegative.py:186]: It looks like `proc` is defined _below_ this statement? Is this a bug?

File: Fitter/hist_files/selectedWCs.txt (hunk 1 above)
Comment (sscruz) [reply Fitter/hist_files/selectedWCs.txt:1]: this is generated by the datacard producer per each region. it happened that the jsons for all the regions were the same. I will rename it to `.json`

File: Fitter/python/AnomalousCouplingEFTNegative.py (hunk 2 above)
Comment (sscruz) [reply Fitter/python/AnomalousCouplingEFTNegative.py:186]: not sure I understand, here we are calling `'proc'`, not `proc`...

Comment (bryates) [reply Fitter/python/AnomalousCouplingEFTNegative.py:186]: Oh I looked passed the quotes!

**Pull request merged**